### PR TITLE
feat(ap-1989): communicating by message

### PR DESCRIPTION
**Pull request by A-Kasaaian**
I have added message to replace it with sessionStorage. There was a mixture of different storages used to pas the same variable around, I modified it to use only 1 storage in extension..

There was an assumption that we need integration to inject the code to customer page, but I made changes to inject a new file into the page that gives us access to local variables of the page. We now have access to evolv context from there.

### CODE REVIEW

File: contentScript.js [hunk 1]
```diff
@@ -1,4 +1,10 @@
-window.sessionStorage.setItem('evoToolsEnabled', true);
+const injectScript = () => {
+    var script = document.createElement('script');
+    script.src = chrome.runtime.getURL('injectScript.js');
+    // script.setAttribute('data-exid', chrome.runtime.id);
```
Comment (robertsevernsentient) [inline contentScript.js]: This commented out stuff needed?

File: contentScript.js [hunk 2]
```diff
@@ -62,11 +64,11 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
             window.dispatchEvent(new Event('locationchange'));
             break;
         case 'enable_evolv':
-            window.sessionStorage.removeItem('evolv:blockExecution');
+            chrome.storage.sync.remove('evolv:blockExecution');
```
Comment (robertsevernsentient) [inline contentScript.js:64]: Have you verified this behavior? We do prevent the snippet running by setting blockeExecution on the session

File: injectScript.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        window.runEvotoolsContentScript = true;
+    }, 0);
+};
+
+poll();
+
+/**	
+ * Handle SPA transitions	
+ */
+window.addEventListener('locationchange', function () {
```
Comment (robertsevernsentient) [inline injectScript.js:41]: I know this is existing code being moved around -- but I don't think we need this.

We have monkey patching for this in asset manager when the config is enabled.

I think this is trying to get round updataing the remote context on page load. We can just do it with a listener

File: popup.js [hunk 4]
```diff
@@ -43,17 +44,6 @@ const setUidValue = () => {
   });
 };
```
Comment (robertsevernsentient) [inline popup.js:46]: Was environment id missing in the demo today?

File: injectScript.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        window.evolv.context.remoteContext.experiments !== undefined &&
+        window.evolv.context.remoteContext.experiments.allocations !== undefined
+}
+
+var pollingSafetyNet = 0;
+function poll() {
+    if (pollingSafetyNet++ < 50) {
+        if (!isEvolvLoaded()) return setTimeout(poll, 100);
+
+        window.postMessage({
```
Comment (robertsevernsentient) [inline injectScript.js:14]: So here is where I would switch to the listener approach
on context.changed
https://media.evolv.ai/javascript-sdk/releases/latest/docs/EvolvClient.html#on

send the remote context to the extension

**Review by robertsevernsentient — APPROVED** [review]

File: contentScript.js (hunk 1 above)
Comment (A-Kasaaian) [reply contentScript.js]: Not now but we may need it later.

Comment (A-Kasaaian) [reply contentScript.js]: I remove them to keep clean will implement later if needed.

File: popup.js (hunk 4 above)
Comment (A-Kasaaian) [reply popup.js:46]: No and it's not working in master too.

File: contentScript.js (hunk 2 above)
Comment (A-Kasaaian) [reply contentScript.js:64]: Good to know. This button doesn't work. I need to make changes in the code to make it work. I have to revert this change as well.

**Pull request merged**